ir/src/main/scala/org/scalajs/core/ir/Hashers.scala

```diff
@@ -53,8 +53,29 @@ object Hashers {
     val size = 2 * (if (considerPos) 2 else 1) * 20
     val builder = new StringBuilder(size)
 
-    def append(hash: Array[Byte]) =
-      for (b <- hash) builder.append(f"$b%02X")
+    def hexDigit(digit: Int): String = {
+      digit match {
+        case 0  => "0"
+        case 1  => "1"
+        case 2  => "2"
+        case 3  => "3"
+        case 4  => "4"
+        case 5  => "5"
+        case 6  => "6"
+        case 7  => "7"
+        case 8  => "8"
+        case 9  => "9"
+        case 10 => "A"
+        case 11 => "B"
+        case 12 => "C"
+        case 13 => "D"
+        case 14 => "E"
+        case _  => "F"
+      }
+    }
```

Comment:
That is the best solution, the time got down to:
```
[debug] Linker: Assemble LinkedClasses: 40811 us
```
I tried returning a `Char`, it also improved it, but not as much as this. I also tried the indexed array, but it was slower than the `switch`.